From 375cb812fc7c06c40f1b523d12d7ee52fb146577 Mon Sep 17 00:00:00 2001
From: Simmo Saan <simmo.saan@gmail.com>
Date: Tue, 25 Oct 2022 03:22:36 +0300
Subject: [PATCH 14/16] Fix compilation on OCaml 5.0.0~beta1 (#1094)

* Add BatBytes.unsafe_escape on OCaml 5

* Add BatPrintexc.string_of_extension_constructor on OCaml 5
---
 src/batBytes.mliv    | 2 ++
 src/batPrintexc.mliv | 9 +++++++++
 2 files changed, 11 insertions(+)

diff --git a/src/batBytes.mliv b/src/batBytes.mliv
index c9881e14..e6cd824e 100644
--- a/src/batBytes.mliv
+++ b/src/batBytes.mliv
@@ -766,3 +766,5 @@ let s = Bytes.of_string "hello"
 ##V>=4.4##external unsafe_fill:  t -> int -> int -> char -> unit = "caml_fill_bytes" [@@noalloc]
 
 ##V>=4.09##external unsafe_blit_string : string -> int -> t -> int -> int -> unit = "caml_blit_string" [@@noalloc]
+
+##V>=5.0##val unsafe_escape : bytes -> bytes
diff --git a/src/batPrintexc.mliv b/src/batPrintexc.mliv
index 39bedf91..0ef92bb6 100644
--- a/src/batPrintexc.mliv
+++ b/src/batPrintexc.mliv
@@ -416,3 +416,12 @@ val print : _ BatInnerIO.output -> exn -> unit
 ##V>=4.12##
 ##V>=4.12##    @since 3.3.0 and 4.12
 ##V>=4.12##*)
+
+(**/**)
+
+(**  {1 Obj printer}
+  Unsafe printer used in this module and exposed to the other standard library
+   module
+*)
+
+##V>=5.0##val string_of_extension_constructor: Obj.t -> string
-- 
2.38.1

